app/controllers/users_controller.rb

```diff
@@ -73,17 +51,22 @@ def update_password(user)
 		end
 	end
 
-	def save_profile_pic(file)
-		user_img_dir_path = Rails.root.join('public', 'uploads', current_user.id.to_s)
-		Dir.mkdir(user_img_dir_path) unless Dir.exist?(user_img_dir_path)
-	
-		begin
-			File.open(user_img_dir_path.join(file.original_filename), 'wb') do |f|
-				f.write(file.read)
-			end
-		rescue IOError => e
-			puts "IOError occurred: #{e.message}"
+	def require_login
+		unless current_user
+			flash[:danger] = "You must be logged in to view this page."
```

Comment:
Use respond_with in case this is a JSON request

https://apidock.com/rails/v4.0.2/ActionController/MimeResponds/respond_with

Reply:
Couldn't completely get the respond_with to work (also looked into respond_to) but ended up changing any flashes and redirects to render json.

Reply:
My update action in the UsersController still needs a lot of work but it's functional right now.